File: snippets/js/node/src/gossip.ts
// ANCHOR: gossip
import {
    ClientBuilder,
    NostrGossip,
    RelayCapabilities,
    RelayUrl,
} from "@nostrdevkit/nostr-sdk-node";

async function buildGossipClient() {
    const gossip = await NostrGossip.sqlite("./data/gossip.sqlite");
    const client = new ClientBuilder().gossip(gossip).build();
    const capabilities = RelayCapabilities.discovery();

    await client.addRelay(RelayUrl.parse("wss://relay.damus.io"), capabilities, false, undefined);
    await client.addRelay(RelayUrl.parse("wss://purplepag.es"), capabilities, false, undefined);
    await client.connect(undefined);

    return client;
}
// ANCHOR_END: gossip

await buildGossipClient();

File: snippets/python/src/gossip.py
import asyncio

# ANCHOR: gossip
from nostr_sdk import (
    Client,
    ClientBuilder,
    NostrGossip,
    RelayCapabilities,
    RelayUrl,
)


async def build_gossip_client() -> Client:
    gossip = await NostrGossip.sqlite("./data/gossip.sqlite")
    client = ClientBuilder().gossip(gossip).build()
    capabilities = RelayCapabilities.discovery()

    await client.add_relay(
        RelayUrl.parse("wss://relay.damus.io"), capabilities=capabilities
    )
    await client.add_relay(
        RelayUrl.parse("wss://purplepag.es"), capabilities=capabilities
    )
    await client.connect()

    return client
# ANCHOR_END: gossip


if __name__ == "__main__":
    asyncio.run(build_gossip_client())
